src/wallet/wallettool.cpp

```diff
@@ -147,7 +126,20 @@ bool ExecuteWalletToolFunc(const std::string& command, const std::string& name)
             WalletShowInfo(wallet_instance.get());
             wallet_instance->Flush(true);
         } else if (command == "salvage") {
-            return SalvageWallet(path);
+            bilingual_str error;
+            std::vector<bilingual_str> warnings;
+            bool ret = RecoverDatabaseFile(path, error, warnings);
+            if (!ret) {
+                if (warnings.size() > 0) {
+                    for (const auto warning : warnings) {
+                        tfm::format(std::cerr, "%s\n", warning.original);
```

Comment:
style nit in commit "wallettool: ..."

A range based for loop does not need to be guarded by a size check, so you can remove the `if (warnings.size() > 0)`

Reply:
Fixed.